app/views/profiles/search.html.erb

```diff
@@ -3,13 +3,13 @@
  
   <fieldset>
     <%= label(:gender, :male) %>
-    <%= radio_button_tag 'gender', 0 %>
+    <%= radio_button_tag 'search[gender]', 0 %>
```

Comment:
See if you can make these references dynamic instead of hard-coded. I think there is a way of getting all the possible enum values from the model.